### ISSUE TITLE
E (2843) chip[SVR]: Error when trying to join Group 257 of fabric index 1 : 10e

### ISSUE THREAD

**Issue by nykh2010** [project-chip/connectedhomeip#18984]
when I use "./chip-tool groups add group 257 "grp" 1 0" to add esp32 to one group. it is fail when join the group. the error log is "E (2843) chip[SVR]: Error when trying to join Group 257 of fabric index 1 : 10e". 
I use the lighting-app example.
[multicastAddress.log](https://github.com/project-chip/connectedhomeip/files/8803315/multicastAddress.log)

**Comment by mrjerryjohns** [comment]
`10e` is an `INET_ERROR_UNKNOWN_INTERFACE`.

**Comment by mrjerryjohns** [comment]
The above error is being emitted from `Server::RejoinExistingMulticastGroups`, when trying to call `mTransports.MulticastGroupJoinLeave`. Someone versed in group comms should help take a look at this...

**Comment by bzbarsky-apple** [comment]
@dhrishi @mkardous-silabs @turon

I expect that `UDPEndPointImplLwIP::IPv6JoinLeaveMulticastGroupImpl` or `UDPEndPointImplLwIP::IPv4JoinLeaveMulticastGroupImpl` is what's returning INET_ERROR_UNKNOWN_INTERFACE, when `FindNetifFromInterfaceId` returns null.  

Looks like `UDP::MulticastGroupJoinLeave` does:
```
    return mUDPEndPoint->LeaveMulticastGroup(mUDPEndPoint->GetBoundInterface(), address.GetIPAddress());
```
and I am wondering whether `mUDPEndPoint->GetBoundInterface()` is the "wildcard" interface id, in that the UDP endpoint is not bound to a specific interface at all.... I am willing to bet it is.

If that's what's going on, then that's also the thing that breaks group comms on Mac...

**Comment by jepenven-silabs** [comment]
No work was done on our side for LwIP support of Multicast. There was a compatibility issue between the OpenThread Stack and the LwIP stack fro groups on top of code duplication hence we move aside from using LwIP and that context.

**Comment by dtodor** [comment]
We are having the same issue. Is group messaging supposed to work on EFR32 and ESP32?

**Comment by woody-apple** [comment]
Spec Issue Review: @rcasallas-silabs Are you able to take a look at this group issue?

**Comment by jepenven-silabs** [comment]
This is platform specific to LwIP, issue is not present on Native Openthread platform nor on Zephyr platform. @Damian-Nordic can you confirm that it indeed works on Zephyr?

De-assigning Ricardo from this issue since its platform specific

**Comment by jepenven-silabs** [comment]
More over if the device is using IPV4 this is expected behavior at that point since multicast doesn't support IPV4

**Comment by Damian-Nordic** [comment]
@jepenven-silabs Yes, it definitely worked the last time I tried it on Zephyr platform and I believe we have automated tests for this functionality.

**Comment by andy31415** [comment]
Looked a bit at 20136 - I believe our current code only uses mld6_joingroup_netif and never mld6_joingroup
For wildcard inet interface, we probably need to use mld6_joingroup

What I am unclear about is socket implementation ... there interface index 0 means `default interface` and I am unclear what default is. However if sockets work now, maybe we only need to fix lwip.

**Comment by andy31415** [comment]
If group comms is also broken on mac, we may need group communication to use a separate endpoint for every interface (minmdns does this currently, but it also needed this to be able to send and receive unicast packets I believe).

**Comment by woody-apple** [comment]
Cert Blocker Review: Closing this, given we believe this is resolved.
